### PR TITLE
doc: setup sphinx infrastructure for doc build

### PR DESCRIPTION
**Pull request by dbkinder**
This only sets up the infrastructure for building documentation using Sphinx.  We'll continue with documentation sections in further PRs.

The publishing to our opea-project.github.io site will also come later as we have that content ready to publish.

### CODE REVIEW

File: scripts/filter-doc-log.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# Copyright (C) 2019-2022 Intel Corporation.
```
Comment (kevinintel) [inline scripts/filter-doc-log.sh:2]: # Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

File: scripts/filter-known-issues.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
+#! /usr/bin/env python3
+# Copyright (c) 2017-2022, Intel Corporation
```
Comment (kevinintel) [inline scripts/filter-known-issues.py:2]: # Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

**Comment by ftian1** [comment]
David, is there a render version for us to check? I tried to build html in my local with sphinx-build, but fails with python 3. looks like the "from version import __version__" is not workable in python 3. 

and another comment is shall we create a new folder like "sphinx" under this root dir to store all related stuff. current PR is putting every thing (404.rst and other docs) in root dir. it may be a little confusing for user.

**Comment by chensuyue** [comment]
Suggest to put all the IO page build related code into one folder.

**Comment by dbkinder** [comment]
> David, is there a render version for us to check? I tried to build html in my local with sphinx-build, but fails with python 3. looks like the "from version import \_\_version\_\_" in config.py is not workable in python 3. 
> 
> 
> 
> and another comment is shall we create a new folder like "sphinx" under this root dir to store all related stuff. current PR is putting every thing (404.rst and other docs) in root dir. it may be a little confusing for user.

I'm preparing a doc build document that explains the setup needed, including using scripts/requirements.txt for python packages. For example, you'll need the other four opea-project repos cloned as sisters to the cloned docs repo (version.py lives in the GenAIComps repo). 

I've used this same directory layout for other projects using Sphinx. The docs repo is for building the published docs, but I suppose I could move some files into a doc_build directory and update things accordingly.

File: index.rst [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.. _acrn_home:
+
+OPEA Project Documentation
+##########################
+
+Welcome to the OPEA Project (|version|) documentation published |today|.
+OPEA streamlines implementation of enterprise-grade Generative AI by efficiently
+integrating secure, performant, and cost-effective Generative AI workflows into business value.
```
Comment (mkbhanda) [inline index.rst:8]: s/into/to deliver

File: scripts/filter-doc-log.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# SPDX-License-Identifier: BSD-3-Clause
+
+# run the filter-known-issues.py script to remove "expected" warning
+# messages from the output of the document build process and write
+# the filtered output to stdout
+#
+# Only argument is the name of the log file saved by the build.
```
Comment (mkbhanda) [inline scripts/filter-doc-log.sh:9]: s/saved by the build/to save the build log.

**Review by mkbhanda — APPROVED** [review]
2 minor comments.

File: scripts/filter-doc-log.sh (hunk 1 above)
Comment (dbkinder) [reply scripts/filter-doc-log.sh:2]: Thanks, updated.

File: scripts/filter-known-issues.py (hunk 2 above)
Comment (dbkinder) [reply scripts/filter-known-issues.py:2]: Thanks, updated.

File: index.rst (hunk 3 above)
Comment (dbkinder) [reply index.rst:8]: I grabbed that text from https://opea.dev/ :)

**Comment by dbkinder** [comment]
> David, is there a render version for us to check? I tried to build html in my local with sphinx-build, but fails with python 3. looks like the "from version import __version__" in config.py is not workable in python 3.
> 
> and another comment is shall we create a new folder like "sphinx" under this root dir to store all related stuff. current PR is putting every thing (404.rst and other docs) in root dir. it may be a little confusing for user.

I've got this document available for how to setup for doing a doc build:  https://opea-project.github.io/latest/developer-guides/docbuild.html

**Comment by dbkinder** [comment]
@ftian1 @chensuyue @hshen14  I could really use your approval and merge for this initial setup so I can continue with fleshing out the initial documentation.

> Suggest to put all the IO page build related code into one folder.

I've moved most of the sphinx-specific stuff into a sphinx directory now.  Some really needs to be where it currently is.

**Comment by ftian1** [comment]
> > David, is there a render version for us to check? I tried to build html in my local with sphinx-build, but fails with python 3. looks like the "from version import **version**" in config.py is not workable in python 3.
> > and another comment is shall we create a new folder like "sphinx" under this root dir to store all related stuff. current PR is putting every thing (404.rst and other docs) in root dir. it may be a little confusing for user.
> 
> I've got this document available for how to setup for doing a doc build: https://opea-project.github.io/latest/developer-guides/docbuild.html

thanks for the great work. the render version looks very good to me.

**Review by ftian1 — APPROVED** [review]

**Pull request merged**